service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/CodeableConceptCdMapper.java

```diff
@@ -316,12 +317,16 @@ private List<Coding> getNonSnomedCodeCodings(CodeableConcept codeableConcept) {
             .filter(coding -> !isSnomed(coding))
             .toList();
 
+        List<Coding> nonSnomedCodings = new ArrayList<>();
+
         for (Coding coding : nonSnomedCodeCodings) {
             var hl7CodeSystem = CodeSystemsUtil.getHl7code(coding.getSystem());
-            coding.setSystem(hl7CodeSystem);
+            if (!hl7CodeSystem.isEmpty()) {
+                nonSnomedCodings.add(new Coding(hl7CodeSystem, coding.getCode(), coding.getDisplay()));
```

Comment:
This was needed as an issue was occurring where the same coding block was processed twice when it was updated whilst iterating the list